app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -7,34 +7,110 @@ module DynamicGraphs.GraphGenerator
 reqsToGraph reqs =
-    let stmts = concatMap reqToStmts reqs
+    let stmts = stmtslst (foldUpReqLst reqs 0)
     in
         buildGraph stmts
 
-reqToStmts :: (Text, Req) -> [DotStatement Text]
-reqToStmts (name, NONE) = [DN $ DotNode name []]
-reqToStmts (_, _) = []
+-- convert the original requirement data into dot statements that can be used by buildGraph to create the
+-- corresponding DotGraph objects. ([] is the [] after name the optional parameters for DotNode?)
+reqToStmts :: (Text, Req) -> StmtsWithCounter -> StmtsWithCounter
+
+
+reqToStmts (name, NONE) stmtcounter = StmtsWithCounter{stmtslst = connectLst (stmtslst stmtcounter) [DN $ DotNode (name `mappend` (pack (show (counter stmtcounter)))) []]
+                                      , counter = (counter stmtcounter) + 1}
+
+
+
+
+reqToStmts (name, J string1) stmtcounter = StmtsWithCounter{stmtslst = connectLst [DE $ DotEdge str1 name []] (stmtslst (foldUpReqLst [(name, NONE), (str1, NONE)] 0))
+                                                           , counter = (counter stmtcounter) + 3}
+  where str1 = pack string1
+
+--reqToStmts (name, AND reqs1) stmtcounter = connectLst [DN $ DotNode "and" [], DE $ DotEdge "and" name []] (connectLst andedges nodestatements)
+--  where andedges = createAndEdge (dragNodeIdInReq reqs1)
+--       nodestatements = concatMap reqToStmts ((name, NONE):(decompJString reqs1))
+
+--reqToStmts (name, OR reqs1) stmtcounter = connectLst [DN $ DotNode "or" [], DE $ DotEdge "or" name []] (connectLst oredges nodestatements)
+--  where oredges = createOrEdge (dragNodeIdInReq reqs1)
+--        nodestatements = concatMap reqToStmts ((name, NONE):(decompJString reqs1))
+
+--reqToStmts (name, FROM string1 (J string2)) stmtcounter = connectLst fromedges (concatMap reqToStmts [(name, NONE), (pack_string1, NONE), (pack_string2, NONE)])
+--  where pack_string1 = pack string1
+--        pack_string2 = pack string2
+--        fromedges = [DE $ DotEdge pack_string2 pack_string1 [], DE $ DotEdge pack_string1 name []]
+
+--reqToStmts (name, GRADE string1 (J string2)) stmtcounter = connectLst gradeedges (concatMap reqToStmts [(name, NONE), (pack string1, NONE), (pack string2, NONE)])
+--  where pack_string1 = pack string1
+--        pack_string2 = pack string2
+--        gradeedges = [DE $ DotEdge pack_string2 pack_string1 [], DE $ DotEdge pack_string1 name []]
+
+--reqToStmts (name, RAW string1) stmtcounter = connectLst rawedges (concatMap reqToStmts [(name, NONE), (pack_string1, NONE)])
+--  where pack_string1 = pack string1
+--        rawedges = [DE $ DotEdge pack_string1 name []]
+
+--reqToStmts (_, _) stmtcounter = []
+
+data StmtsWithCounter = StmtsWithCounter{ stmtslst :: [DotStatement Text]
+                                        , counter   :: Int }
+
+foldUpReqLst :: [(Text, Req)] -> Int -> StmtsWithCounter
+foldUpReqLst reqlst count = foldl(\acc x -> reqToStmts x acc) StmtsWithCounter{stmtslst = [], counter = count} reqlst
+
+
+
```

Comment:
Be careful with blank lines -- generally, there should only be *one* blank line between consecutive definitions.